packages/task/src/browser/task-configurations.ts

```diff
@@ -54,24 +55,24 @@ export class TaskConfigurations implements Disposable {
     protected client: TaskConfigurationClient | undefined = undefined;
 
     constructor(
-        @inject(FileSystemWatcherServer) protected readonly watcherServer: FileSystemWatcherServer,
+        @inject(FileSystemWatcher) protected readonly watcherServer: FileSystemWatcher,
         @inject(FileSystem) protected readonly fileSystem: FileSystem
     ) {
         this.toDispose.push(watcherServer);
-
-        watcherServer.setClient({
-            onDidFilesChanged: async event => {
-                try {
-                    const watchedConfigFileChanges = event.changes.filter(change => this.watchedConfigFileUris.some(fileUri => fileUri === change.uri));
-                    if (watchedConfigFileChanges.length >= 0) {
-                        await this.onDidTaskFileChange(watchedConfigFileChanges);
-                        if (this.client) {
-                            this.client.taskConfigurationChanged(this.getTaskLabels());
-                        }
+        this.watcherServer.onFilesChanged(async changes => {
+            try {
+                const watchedConfigFileChanges = changes.filter(change => this.watchedConfigFileUris.some(fileUri => fileUri === change.uri.toString()))
```

Comment:
`this.watchedConfigFileUris.filter(fileUri => FileChangeEvent.isAffected(changes, new URI(fileUri)))`  otherwise recursive deletion is not handled properly

Reply:
Thank you for the review. I changed the code to use isAffected() instead.